Editors/vscode/src/inlayHints.ts

```diff
@@ -0,0 +1,201 @@
+'use strict';
+import * as vscode from 'vscode';
+import * as langclient from 'vscode-languageclient/node';
+import { InlayHint, InlayHintsParams, inlayHintsRequest } from './lspExtensions';
+
+// The implementation is loosely based on the rust-analyzer implementation
+// of inlay hints: https://github.com/rust-analyzer/rust-analyzer/blob/master/editors/code/src/inlay_hints.ts
+
+// Note that once support for inlay hints is officially added to LSP/VSCode,
+// this module providing custom decorations will no longer be needed!
+
+export async function activateInlayHints(
+    context: vscode.ExtensionContext,
+    client: langclient.LanguageClient
+): Promise<void> {
+    const config = vscode.workspace.getConfiguration('sourcekit-lsp');
+    let updater: HintsUpdater | null = null;
+
+    const onConfigChange = async () => {
+        const wasEnabled = updater !== null;
+        const isEnabled = config.get<boolean>('inlayHints.enabled', true);
+
+        if (wasEnabled !== isEnabled) {
+            updater?.dispose();
```

Comment:
Since we are throwing away the old instance either way, I believe we always need to dispose it.

Reply:
But in the `if`-branch we have `isEnabled == true`, thus `wasEnabled == false` and thus `updater === null`. Hence `updater?.dispose()` doesn’t do anything. I mean, it doesn’t do any harm, but it confused me for a bit.

Reply:
Ah yes, you are right. Probably a question of style, but keeping it this way feels cleaner to me, especially since you can reason about the block locally (i.e. `updater` will always get disposed correctly, regardless of the surrounding condition).